homeassistant/components/sensor/vera.py

```diff
@@ -0,0 +1,165 @@
+"""
+Support for Vera sensors.
+
+Configuration:
+To use the Vera sensors you will need to add something like the following to
+your config/configuration.yaml
+
+sensor:
+    platform: vera
+    vera_controller_url: http://YOUR_VERA_IP:3480/
+    device_data:
+        12:
+            name: My awesome sensor
+            exclude: true
+        13:
+            name: Another sensor
+
+VARIABLES:
+
+vera_controller_url
+*Required
+This is the base URL of your vera controller including the port number if not
+running on 80
+Example: http://192.168.1.21:3480/
+
+
+device_data
+*Optional
+This contains an array additional device info for your Vera devices.  It is not
+required and if not specified all sensors configured in your Vera controller
+will be added with default values.  You should use the id of your vera device
+as the key for the device within device_data
+
+These are the variables for the device_data array:
+
+name
+*Optional
+This parameter allows you to override the name of your Vera device in the HA
+interface, if not specified the value configured for the device in your Vera
+will be used
+
+
+exclude
+*Optional
+This parameter allows you to exclude the specified device from homeassistant,
+it should be set to "true" if you want this device excluded
+
+"""
+import logging
+import time
+from requests.exceptions import RequestException
+
+from homeassistant.helpers import Device
+from homeassistant.const import (
+    ATTR_BATTERY_LEVEL, ATTR_TRIPPED, ATTR_ARMED, ATTR_LAST_TRIP_TIME)
+# pylint: disable=no-name-in-module, import-error
+import homeassistant.external.vera.vera as veraApi
+
+_LOGGER = logging.getLogger(__name__)
+
+
+# pylint: disable=unused-argument
+def get_devices(hass, config):
+    """ Find and return Vera Sensors. """
+
+    base_url = config.get('vera_controller_url')
+    if not base_url:
+        _LOGGER.error(
+            "The required parameter 'vera_controller_url'"
+            " was not found in config"
+        )
+        return False
+
+    device_data = config.get('device_data', {})
+
+    vera_controller = veraApi.VeraController(base_url)
+    categories = ['Temperature Sensor', 'Light Sensor', 'Sensor']
+    devices = []
+    try:
+        devices = vera_controller.get_devices(categories)
+    except RequestException as inst:
+        # There was a network related error connecting to the vera controller
+        _LOGGER.error("Could not find Vera sensors: %s", inst)
```

Comment:
If you do `_LOGGER.exception("Error communicating with Vera API")` then the logger will automatically print out the exception including stack trace